### ISSUE TITLE
Is the plan to compile exclusively with PGI on Drone?

### ISSUE THREAD

**Issue by bast** [bast/cmake-recipes#214]
Because if yes, then the menu.yml files seem a bit inconsitent and we could avoid many of the local menu.yml files and put stuff into the global one. But possibly I misunderstood things.

**Comment by robertodr** [comment]
Ideally yes, but in practice not all recipes work with PGI without some changes.

**Comment by bast** [comment]
OK - should I turn on PGI globally on Drone and then we debug the recipes as second step?

**Comment by robertodr** [comment]
I did that already and my "debug" was the mix of PGI/GNU you see now 😸 I didn't try particularly hard to fix the recipes to make them work with PGI. The problems I found were:
1. `pgfortran` does not understand some flags like `-std=f2008`. It seems the highest possible implemented standard is used anyway.
2. The C and C++ compilers are... different. The Eigen recipe will definitely not compile with PGI. That has been a known (and _wontfix_) problem for the Eigen folks.
3. Mixing libraries built with GNU (like HDF5) or using BLAS/LAPACK with the PGI/GNU setup leads to mayhem.

Basically we need to decide how hard we want to work on the recipes to get full PGI support. I think the current effort is enough and is based on the very pragmatic consideration that PGI seems to be used a lot by Fortran projects and that the book is not exclusively about Fortran.

**Comment by bast** [comment]
OK I see and I agree. But should the "global" menu.yml not be "pure" PGI and we "fix" problems locally inside recipes? By having a global PGI which is locally fixed with GNU the message of this test server might be more clear.

**Comment by robertodr** [comment]
Yes, that makes for sense.

**Comment by bast** [comment]
Wouldn't it be more relevant to test with Intel or is this tricky license-wise?

**Comment by robertodr** [comment]
I am revamping the docker images. Intel could be tricky license-wise, but I
can look into that.

**Comment by bast** [comment]
Plan is to test Intel and PGI.
